### DISCUSSION TITLE
ID and Relay server setting does not belong to the client settings

### DISCUSSION THREAD

**Discussion by attikov** [rustdesk/rustdesk#2913] [General]
ID and Relay server setting does not belong to the clident settings but to the 
settings of the remote machine.

The currency design makes no sense as I can't manage remotes from my self hosted server
and other ones together.

thx

**Comment by bigretromike** [comment]
If you are talking about stable version, and that the ID server and Relays server can be configure by user but not by remote hand, then fear not, that is fixed in daily build with UAC being require to access those settings and you can configure host to give you access to those setting via remote access.

**Comment by attikov** [comment]
I don't understand a word you are talking about.

I'm telling that ID/Relay server setting should be per remote configurable and not global as I can't connect to remotes outside of this global setting.

**Comment by bigretromike** [comment]
Server settings are and should be global as the service is installed globally. Also if its not global you wouldn't be able to connect to not log in machine, as you can now (unattended machine).

**Comment by attikov** [comment]
You are mixing things up.
There is a difference between a "default value" and a "global setting".
The ID and Relay server setting should be configurable per contact and there can be a default setting with the ID/Relay server provided by RustDesk.
Right now, if I set ID/Relay server for my own one, I can't connect anymore to remotes not having the same ID/Relay server set up. This should be solved by a "per contact" setting, not a global one.

**Comment by bigretromike** [comment]
Oh, sorry.
But then, maybe we should have working Address Book instead of relaying on "last connected" ? that would help even more as you could just segragate those contacts via groups and maybe set relay server on the group instead of per contact

**Reply by pedjas** [reply]
Problem is more complex. If you have installed client for one server then you cannot have client of r other server. All RustDesk clients installed on one machine save their configuration to the same place, fighting each other.

So far, the only solution is using so called portable mode, but that does not resolve issues 100%, as that is not real potable mode, multiple instances still save configurations at the same place.

This is reported long time ago. It does not seem to be resolved in near future.

**Reply by SergeyMy** [reply]
if the client is NOT installed in the system - it stores all settings ONLY in the user profile

**Comment by attikov** [comment]
Or both. But I don't want to make suggestions regarding the implementation details.

**Comment by bigretromike** [comment]
here is the address book / contact list issue, you can +1 there so maybe its get re-implemented faster: https://github.com/rustdesk/rustdesk/issues/177

**Comment by attikov** [comment]
Done. But as I said it, I'm not into the implementation details and this does not necessarily needs an address book nor does an addressbook alone solves this.

**Comment by nickdickinson** [comment]
I agree @attikov. This is not about an address book implementation as far as I can tell. The way I would explain this (if I understand the use case) is that I want to connect to some computer that is not within my org and does not use my relay server. Do do that, I have to delete the relay server and key fields unfortunately. So it means each time I switch between the computers either in or outside my org that do not use the relay server, I have to keep changing these fields.

**Comment by SergeyMy** [comment]
> ...So it means each time I switch between the computers either in or outside my org that do not use the relay server, I have to keep changing these fields.

and it's wonderful! organization security :)
separate user for the organization, separate for external.
the ability to connect to someone - opens the possibility of connecting to you!

**Comment by nickdickinson** [comment]
So far this discussion which does not seem to be moving towards resolving this use case within rustdesk by enabling a user to define different relay servers and keys per remote connection (for a one-way connection - not per se allowing connecting back unless your servers are the same).  The workaround that I'm deriving from the comments is to create a new system level user for each remote you need to connect to and to ONLY use the portable version and run only one at a time. Is that correct?

Related to #1258 which may also be a solution.

**Reply by pedjas** [reply]
I tested this and my conclusion is that you can use same Windows account but you must not use install version of RustDesk. 

If you have several portable versions, those would work as long as one is run at a time.

If you have installed version it would always override settings of portables and those would not work.

**Reply by bigretromike** [reply]
You could also use "shortcuts" to create a "contact" like file so you double click it and rust will connect to that ID, put them in folders (organizations) and add some metadata so you know you need to "switch" or add that to portable versions with custom servers :-)

**Reply by nickdickinson** [reply]
Clever workaround with shortcuts / shell or batch scripts. I guess the next level would be to first check if there is already a rustdesk process running and ask the user to stop it.

**Reply by pedjas** [reply]
Next level would be if Rustesk is fixed to save portable configurations in folder where exe is located, so there is no overriding each other configurations and multiple instances could run at the same time.

That is literally three lines of code change.

**Reply by nickdickinson** [reply]
@pedjas have you tested the code? Pull request?

**Reply by pedjas** [reply]
What code? I tested the latest official release, and also some of the Alpha releases, not recent.

**Comment by Slydder** [comment]
We have just decided to not use the relay server at all until this BADLY needed feature is added. we connect to too many different systems daily to be messing around with different mobile instances and a buttload of shortcuts. cannot, for the life of me, see how this was not thought of in advance.

strange. hope to see it corrected soon though. otherwise the best remote desktop software on the market.
